x-pack/plugins/security_solution/server/lib/detection_engine/prebuilt_rules/logic/rule_asset/rule_asset_saved_objects_client.ts

```diff
@@ -27,33 +26,53 @@ export interface IRuleAssetSavedObject {
   attributes: IRuleAssetSOAttributes;
 }
 
-export interface RuleAssetSavedObjectsClient {
-  find: (
-    options?: Omit<SavedObjectsFindOptions, 'type'>
-  ) => Promise<SavedObjectsFindResponse<IRuleAssetSavedObject>>;
-  all: () => Promise<IRuleAssetSavedObject[]>;
+export interface RuleAssetsClient {
+  fetchLatestVersions: () => Promise<IRuleAssetSOAttributes[]>;
 }
 
-export const ruleAssetSavedObjectsClientFactory = (
+export const ruleAssetsClientFactory = (
   savedObjectsClient: SavedObjectsClientContract
-): RuleAssetSavedObjectsClient => {
+): RuleAssetsClient => {
   return {
-    find: (options) =>
-      savedObjectsClient.find<IRuleAssetSavedObject>({
-        ...options,
-        type: ruleAssetSavedObjectType,
-      }),
-    all: async () => {
-      const finder = savedObjectsClient.createPointInTimeFinder({
-        perPage: DEFAULT_PAGE_SIZE,
-        type: ruleAssetSavedObjectType,
+    fetchLatestVersions: () => {
+      return withSecuritySpan('RuleAssetsClient.fetchLatestVersions', async () => {
+        const findResult = await savedObjectsClient.find<
+          IRuleAssetSavedObject,
+          {
+            rules: AggregationsMultiBucketAggregateBase<{
+              latest_version: AggregationsTopHitsAggregate;
+            }>;
+          }
```

Comment:
I didn't know that it's possible to type aggregations for the `find` method using some built-in types. Very useful!